src/isomorphic/children/__tests__/ReactChildren-test.js

```diff
@@ -850,4 +855,45 @@ describe('ReactChildren', () => {
         'to render a collection of children, use an array instead.',
     );
   });
+
+  if (ReactDOMFeatureFlags.useFiber) {
+    describe('with fragments enabled', () => {
+      beforeEach(() => {
+        ReactFeatureFlags = require('ReactFeatureFlags');
+        ReactFeatureFlags.disableNewFiberFeatures = false;
+      });
+
+      it('warns for keys for arrays of elements in a fragment', () => {
+        spyOn(console, 'error');
+        class ComponentReturningArray extends React.Component {
+          render() {
+            return [<div />, <div />];
+          }
+        }
+
+        ReactTestUtils.renderIntoDocument(
+          React.createElement(ComponentReturningArray),
```

Comment:
Let's use JSX here so it's consistent with the [other test](https://github.com/facebook/react/pull/9445/files#diff-48f61d2d3c0e745f59eed2d42b2f833bR893).